modules/third-party/org-apache-felix-fileinstall/patches/LPS-96061.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/org/apache/felix/utils/properties/InterpolationHelper.java b/org/apache/felix/utils/properties/InterpolationHelper.java
+index 5e61bf625d92..928ab15d8142 100644
+--- a/org/apache/felix/utils/properties/InterpolationHelper.java
++++ b/org/apache/felix/utils/properties/InterpolationHelper.java
+@@ -18,6 +18,8 @@ package org.apache.felix.utils.properties;
+ 
+ import java.util.HashMap;
+ import java.util.Map;
++import java.util.regex.Matcher;
++import java.util.regex.Pattern;
+ 
+ import org.osgi.framework.BundleContext;
+ 
+@@ -39,6 +41,8 @@ public class InterpolationHelper {
+     private static final String DELIM_STOP = "}";
+     private static final String MARKER = "$__";
+     private static final String ENV_PREFIX = "env:";
++    private static final Pattern ESCAPED_OPENING_CURLY = Pattern.compile("\\\\+\\{");
++    private static final Pattern ESCAPED_CLOSING_CURLY = Pattern.compile("\\\\+\\}");
+ 
+ 
+     /**
+@@ -407,7 +411,10 @@ public class InterpolationHelper {
+     private static String unescape(String val)
+     {
+         val = val.replaceAll("\\" + MARKER, "\\$");
+-        int escape = val.indexOf(ESCAPE_CHAR);
++        if (!val.matches(".*\\$\\\\*\\{.*\\}.*")) {
```

Comment:
@marianoalvarosaiz can you pull this ".*\\$\\\\*\\{.*\\}.*" out as a Pattern, and resend, thanks!